File: Basic-Cpp/2-Factorial.cpp
/*
Count the number of trailing 0s (number of 0s at the end of the number) in a
factorial number.

Example: Input: 5, Output: 1; Input: 10, Output: 2.
*/

#include <iostream>
#include <cstdlib>
#include <cmath>
using namespace std;

int Factorial(int number){
    int zeros = 0;
    
    for (int i = 1; i != number; i++){
        zeros += number/pow(5,i);
    }
    
    return zeros;
}

int main() {
    int input;
    cout << "Enter a number: ";
    cin >> input;
    cout << "Output: " << Factorial(input) << endl;
    
    return 0;
}

File: Basic-Py/2-Factorial.py
# Week 1 - 2) Count the number of trailing 0s in a factorial number

def Factorial(N):
    ''' Calculates the number of trailing zeros in a factorial. '''

    # Sets noZeros as 0
    noZeros = 0
    factorial = 1
    # Loops in for all the values in range between 1 and the number entered.
    for i in range(1, N):

        # Increments the noZeros by the integer of the number entered divided
        # by 5 to the power.
        noZeros += int(N//(5**i))
        
    return noZeros

if __name__ == "__main__":
    while True:
        try:
            # Allows user to enter a numerical value.
            number = int(input("Enter a number: "))

            # Returns the number of trailing 0s the factorial will have.
            print("Number of Trailing 0s: "+str(Factorial(number)))
        except:
            break
